hangman/lib/hangman/server.ex

```diff
@@ -0,0 +1,27 @@
+defmodule Hangman.Server do
```

Comment:
# -10

Argh, I hate doing this, but your process structure is wonky.

You're supervising the API module, which kicks off a separate server process. This makes things too complex, and also missing the idea that the init/1 function here should really just create the game:

~~~ elixir
def init([]) do
  { :ok, Hangman.Game.new_game() }
end
~~~

This is not just aesthetics. Think of the server's responsibility: it maintains the integrity of the state. And what's the state? An ongoing hangman game.

But the way it is here, the initial server has a state of `[]`. What happens if the client calls make_move or tally before calling new_game?

By setting up the game in `init`, you establish the state's invariant, which you then maintain throughout the life of the server.